replica/table.cc

```diff
@@ -3949,29 +3949,29 @@ future<> compaction_group::cleanup() {
     class compaction_group_cleaner : public row_cache::external_updater_impl {
         table& _t;
         compaction_group& _cg;
+        std::vector<sstables::shared_sstable>& _all_sstables;
         const lw_shared_ptr<sstables::sstable_set> _empty_main_set;
         const lw_shared_ptr<sstables::sstable_set> _empty_maintenance_set;
     private:
         lw_shared_ptr<sstables::sstable_set> empty_sstable_set() const {
             return make_lw_shared<sstables::sstable_set>(_t._compaction_strategy.make_sstable_set(_t._schema));
         }
     public:
-        explicit compaction_group_cleaner(compaction_group& cg)
+        explicit compaction_group_cleaner(compaction_group& cg, std::vector<sstables::shared_sstable>& sstables_to_delete)
                 : _t(cg._t)
                 , _cg(cg)
+                , _all_sstables(sstables_to_delete)
                 , _empty_main_set(empty_sstable_set())
                 , _empty_maintenance_set(empty_sstable_set())
         {
         }
         virtual future<> prepare() override {
             // Capture SSTables after flush, and with compaction disabled, to avoid missing any.
             auto set = _cg.make_sstable_set();
-            std::vector<sstables::shared_sstable> all_sstables;
-            all_sstables.reserve(set->size());
-            set->for_each_sstable([&all_sstables] (const sstables::shared_sstable& sst) mutable {
-                all_sstables.push_back(sst);
+            _all_sstables.reserve(set->size());
+            set->for_each_sstable([this] (const sstables::shared_sstable& sst) mutable {
+                _all_sstables.push_back(sst);
             });
-            co_await _cg.delete_sstables_atomically(std::move(all_sstables));
             if (utils::get_local_injector().enter("tablet_cleanup_failure")) {
```

Comment:
This makes the previous patch (which moves the injection point) have no effect.

Reply:
Right.  The point was that the reproducer uses this error injection point, and it needs it to happen after deleting the sstable (when the patch introducing the reproducer applied before the fix, or the patch with the fix is reverted).  I can annotate that in this patch.